request-manager/api/api.go

```diff
@@ -203,11 +203,8 @@ func (api *API) createRequestHandler(c echo.Context) error {
 	// ----------------------------------------------------------------------
 	// Run (non-blocking)
 
-	// TODO(felixp): if creating the request succeeded but starting it failed,
-	// mark the request as Failed. There's currently no way for a
-	// user to Start a request that's already been created, so otherwise
-	// the request will be Pending forever.
 	if err := api.rm.Start(req.Id); err != nil {
+		api.rm.Fail(req.Id)
```

Comment:
Should check the error returned by Fail here (as a general rule always check the returned error). Since the API is already in the middle of handling another error, I think if an error is returned you should just log it (i.e. save it in the app's logs but don't return the error from the API).

...And of course there's no logging package in the API. So, let's use the same package we use elsewhere in spincycle — import `log "github.com/sirupsen/logrus"` at the top of the file. Then you can call `log.Errorf(...)` with args just like `fmt.Printf(...)` to log the error. For the record, what that does is very similar to just `fmt.Printf`, but most importantly it'll add a timestamp to the output.